dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -0,0 +1,273 @@
+                            .setSeverity(MessageSeverity.ERROR)
+                            .setType(MessageType.SIMPLE_MESSAGE)
+                            .setLife(6000)
+                            .create(), null);
+        } catch (LanguageException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Sets a mapping defined on field variables
+     *
+     * @param indexName - Index where mapping will be updated
+     * @return Collection of fields names whose mapping was set
+     */
+    private static Set<String> addCustomMappingFromFieldVariables(final String indexName) {
+        final FieldFactory fieldFactory = FactoryLocator.getFieldFactory();
+        final Set<String> mappedFields = new HashSet<>();
+
+        try {
+            //Find field variables
+            final List<FieldVariable> fieldVariables = fieldFactory
+                    .byFieldVariableKey(FieldVariable.ES_CUSTOM_MAPPING_KEY);
+
+            for (final FieldVariable fieldVariable : fieldVariables) {
+                Field field = null;
+                ContentType type = null;
+                try {
+                    field = fieldFactory.byId(fieldVariable.fieldId());
+                    type = contentTypeAPI.find(field.contentTypeId());
+                    final JSONObject jsonObject = new JSONObject();
+                    final JSONObject properties = new JSONObject();
+
+                    jsonObject.put(type.variable().toLowerCase(),
+                            new JSONObject()
+                                    .put("properties", new JSONObject()
+                                            .put(field.variable()
+                                                            .toLowerCase(),
+                                                    new JSONObject(fieldVariable.value()))));
+                    properties.put("properties", jsonObject);
+                    esMappingAPI.putMapping(indexName, properties.toString());
+
+                    //Adds to the set the mapped already set for this field
+                    mappedFields.add((type.variable() + StringPool.PERIOD + field.variable())
+                            .toLowerCase());
+
+                } catch (Exception e) {
+                    handleInvalidCustomMappingError(indexName,
+                            type != null ? type.variable() + "." + field.variable() : "[]");
+                    String message = "Error setting custom index mapping from field variable "
+                            + fieldVariable.key();
+
+                    if (field != null) {
+                        message += ". Field: " + field.name();
+                    }
+
+                    if (type != null) {
+                        message += ". Content Type: " + type.name();
+                    }
+
+                    message += ". Custom mapping will be ignored for index: " + indexName;
```

Comment:
Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)